File: app/src/main/java/ru/playsoftware/j2meloader/appsdb/AppDatabase.java
package ru.playsoftware.j2meloader.appsdb;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {AppItem.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

	public abstract AppItemDao appItemDao();

	public static synchronized AppDatabase open(Context context, String dir) {
		return Room.databaseBuilder(
				context.getApplicationContext(),
				AppDatabase.class,
				dir + "/J2ME-apps.db")
				.allowMainThreadQueries()
				.build();
	}
}

File: app/src/main/java/app/dao/AppDatabase.kt
package app.dao

import android.content.Context
import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.Room

@Database(entities = [AppItem::class], version = 2, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun appItemDao(): AppItemDao?

    companion object {
        @JvmStatic
		@Synchronized
        fun open(context: Context, dir: String): AppDatabase {
            return Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                "$dir/J2ME-apps.db"
            ).build()
        }
    }
}